### PR TITLE
Backports for 10.5.1

### PR DESCRIPTION
**Pull request by cjcolvar**
This PR backports most changes to master since 10.5.0 except ones that I felt might be backwards incompatible.  This is in preparation for cutting a 10.5.1 release.

Commits left out of this pull request:
* Updating to use AF12 (56152a6aa3eba04e4155b7eaf74a317fdd7ec447)
* No need to maintain a configuration. ActiveFedora generates it (667d669c695e5f7ee4390e3d70fccbaa2a5f8a3f)
* Use the new Blacklight::AccessControls::SearchBuilder (84700aa677b2b3fbd8e0c6cffd251a4909603bcc)
* Preparing for 11.0.0.rc1 release (e09a819152f1cf6f96fdff528d4baa9e3ff9d541)

@projecthydra/hydra-contributors

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-rvm: 2.4.1
+rvm: 2.4.2
 
 matrix:
   include:
-    - rvm: 2.3.4
-      env: "BLACKLIGHT_VERSION=5.17.2"
+    - env: "RAILS_VERION=5.1.4"
+    - rvm: 2.3.5
+      env: "BLACKLIGHT_VERSION=5.17.2 RAILS_VERSION=5.0.6"
```
Comment (botimer) [inline .travis.yml]: Is this mismatch of RAILS_VERSION intentional?

File: spec/test_app_templates/lib/generators/test_app_generator.rb [hunk 2]
```diff
@@ -3,7 +3,8 @@ class TestAppGenerator < Rails::Generators::Base
   source_root File.expand_path("../../../../spec/test_app_templates", __FILE__)
 
   def install_blacklight
-    generate 'blacklight:install --devise'
+    # we can skip solr because the activefedora installer will also do it (in the hydra:head generator)
```
Comment (botimer) [inline spec/test_app_templates/lib/generators/test_app_generator.rb]: I didn't dig into the exact AF versions... Is this true?

Comment (cjcolvar) [reply spec/test_app_templates/lib/generators/test_app_generator.rb]: I checked and this is true.

File: .travis.yml (hunk 1 above)
Comment (cjcolvar) [reply .travis.yml]: Yes, I believe so.  This change is just a backport of an existing commit on master (which I didn't write).  I was thinking of making a new PR that updates the travis.yml on master which could be backported too if necessary.

Comment (RudyOnRails) [reply .travis.yml]: Sorry not sure if mismatch was spelling or version number but just in case, is VERION correct or should be VERSION?

Comment (botimer) [reply .travis.yml]: Nice catch. Many eyes make... light code review?

Comment (cjcolvar) [reply .travis.yml]: Great catch!

**Pull request merged**